fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/DisbursementRequestServiceImpl.java

```diff
@@ -0,0 +1,123 @@
+                .header("Authorization", credential).build();
+        Gson gson = new GsonBuilder().create();
+        OkHttpClient client = new OkHttpClient();
+        Response response = null;
+        try {
+            response = client.newCall(request).execute();
+            String resObject = response.body().string();
+            if (response.isSuccessful()) {
+
+                JsonObject jsonResponse = JsonParser.parseString(resObject).getAsJsonObject();
+                String accessToken = jsonResponse.get("access_token").getAsString();
+
+                LOG.info("Login to inkomoko Integration  is Successful");
+                return accessToken;
+            } else {
+                LOG.error("Login to inkomoko Integration has failed:" + resObject);
+                throw new LoanDisbursementRequestException("Login to inkomoko Integration has failed", "loan", resObject);
+            }
+        } catch (Exception e) {
+            LOG.error("Login to inkomoko Integration has failed:" + e);
+            throw new LoanDisbursementRequestException("Login to inkomoko Integration has failed", "loan", e);
+        }
+    }
+
+    private String getConfigProperty(String propertyName) {
+        return this.env.getProperty(propertyName);
+    }
+
+    @Override
+    public void disburseRequestLoan(Loan loan, JsonCommand command) {
+        String token = authenticateToIntegrationApi();
+        final ClientOtherInfo clientOtherInfo = this.clientOtherInfoRepository.getByClientId(loan.client().getId());
+        final PaymentTypeData paymentTypes = this.paymentTypeReadPlatformService
+                .retrieveOne(command.longValueOfParameterNamed("paymentTypeId"));
+        // FIXME: 5/12/23 Logic to manage request id incremental and saving to notes/ new table
+        final String requestId = "cbs_" + loan.getId();
+        DisbursementRequestData disbursementRequestData = new DisbursementRequestData(requestId, loan.getAccountNumber(),
+                loan.getPrincpal().getAmount(), loan.getPrincpal().getCurrencyCode(), paymentTypes.getName(),
+                clientOtherInfo.getTelephoneNo(), clientOtherInfo.getBankAccountNumber(), clientOtherInfo.getBankName());
+
+        MediaType JSON = MediaType.get("application/json; charset=utf-8");
+        String json = gson.toJson(disbursementRequestData);
+
+        RequestBody body = RequestBody.create(json, JSON);
+        Request request = new Request.Builder().url(getConfigProperty("fineract.integrations.inkomoko.rest.initiate.disbursement"))
+                .addHeader("Authorization", "Bearer " + token).post(body).build();
+
+        try (Response response = client.newCall(request).execute()) {
+            LOG.info("Received Response from Inkomoko for request  {0} and  loanid {1}  ", loan.getId(), requestId);
+            // FIXME: 5/12/23 Save it to notes or new table part of INKO-240
```

Comment:
@tabrez-fiter  . Is this implementation done  ? . Kindly explain why you do react to the response from Middleware